src/Stratis.Features.SQLiteWalletRepository/SQLiteWalletRepository.cs

```diff
@@ -0,0 +1,695 @@
+            int? walletId = processBlocksInfo.Wallet?.WalletId;
+
+            foreach (Transaction tx in transactions)
+            {
+                // Build temp.PrevOuts
+                uint256 txId = fixedTxId ?? tx.GetHash();
+
+                foreach (TxIn txIn in tx.Inputs)
+                {
+                    if (transactionsOfInterest?.Contains(txIn.PrevOut.Hash, conn, walletId) ?? true)
+                    {
+                        if (prevOuts == null)
+                            prevOuts = TempTable.Create<TempPrevOut>();
+
+                        // We don't know which of these are actually spending from our
+                        // wallet addresses but we record them for batched resolution.
+                        prevOuts.Add(new TempPrevOut()
+                        {
+                            OutputTxId = txIn.PrevOut.Hash.ToString(),
+                            OutputIndex = (int)txIn.PrevOut.N,
+                            SpendBlockHeight = header?.Height ?? 0,
+                            SpendBlockHash = header?.HashBlock.ToString(),
+                            SpendTxIsCoinBase = (tx.IsCoinBase || tx.IsCoinStake) ? 1 : 0,
+                            SpendTxTime = (int)tx.Time,
+                            SpendTxId = txId.ToString(),
+                            SpendTxTotalOut = tx.TotalOut.ToDecimal(MoneyUnit.BTC)
+                        });
+
+                        transactionsOfInterest?.AddTentative(txId);
+                    }
+                }
+
+                // Build temp.Outputs.
+                for (int i = 0; i < tx.Outputs.Count; i++)
+                {
+                    TxOut txOut = tx.Outputs[i];
+
+                    if (txOut.IsEmpty)
+                        continue;
+
+                    if (txOut.ScriptPubKey.ToBytes(true)[0] == (byte)OpcodeType.OP_RETURN)
+                        continue;
+
+                    bool unconditional = transactionsOfInterest?.Contains(txId, conn, walletId) ?? true; // Related to spending details.
+
+                    foreach (Script pubKeyScript in this.GetDestinations(txOut.ScriptPubKey))
+                    {
+                        if (unconditional || (addressesOfInterest?.Contains(pubKeyScript, conn, walletId) ?? true)) // Paying to one of our addresses.
+                        {
+                            // We don't know which of these are actually received by our
+                            // wallet addresses but we records them for batched resolution.
+                            if (outputs == null)
+                                outputs = TempTable.Create<TempOutput>();
+
+                            outputs.Add(new TempOutput()
+                            {
+                                ScriptPubKey = pubKeyScript.ToHex(),              // For matching HDAddress.ScriptPubKey.
+                                RedeemScript = txOut.ScriptPubKey.ToHex(),  // The ScriptPubKey from the txOut.
```

Comment:
This data should go in the `ScriptPubKey` property, and this property should not exist

Reply:
Properly supporting P2SH is not a trivial undertaking. It requires changes to our transaction building APIs, and the redeem scripts need to be safely stored or the P2SH outputs are rendered unspendable. Since the redeem scripts are not derived from the HD seed they are not able to be restored without a complete backup.

Reply:
Both those values are required. The `ScriptPubKey` corresponds to the `ScriptPubKey` value of the `HDAddress` we want to associate the data with. The reason the name `RedeemScript` is used is not to confuse that value with the value in `HDAddress`.

In theory two different `RedeemScript` can contain TxDestination (PKH's) related to the same `HDAddress`. This is why we need to include `RedeemScript` in the primary key of `HDTransactionData`.